index.js

```diff
@@ -312,18 +312,52 @@ function cms(loopbackApplication, options) {
   });
 
 
+  function validateToken(request, callback) {
+    var AccessToken = loopbackApplication.models.AccessToken;
+    var tokenString = request.body.__accessToken || request.query.access_token;
+    AccessToken.findById(tokenString, function(err, token) {
+      if (err || !token) { return callback(err); }
+      return token.validate(callback);
+    });
+  }
+
   /**
    * Save a model hierarchy; req.body contains a model and its relationship data 
    */
   app.put('/model/save', function(req, res) {
-    //TODO: validate access token and ACL
-    var data = req.body;
-    relationalUpsert.upsert(data, function(error, response) {
-      if (error) {
-        res.status(500).send(error);
-      } else {
-        res.send(response);
-      }
+    var AccessToken = loopbackApplication.models.AccessToken;
+    var ACL = loopbackApplication.models.ACL;
+
+    validateToken(req, function(err, isValid) {
+
+      if (err) { return res.status(500).send(err); }
+      if (!isValid) { return res.status(403).send('Forbidden'); }
+
+      var data = req.body;
+
+      AccessToken.findById(data.__accessToken, function(err, token) {
```

Comment:
I would update validateTaken so that it returns (err, isValid, accessToken) so that way we don't need to call AccessToken.findById() here (once being in validateToken).